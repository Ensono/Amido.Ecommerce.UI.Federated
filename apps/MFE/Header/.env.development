ASSETS_PATH=
GENERATE_SOURCEMAP=true
NEXT_BLOB_STORAGE_PATH=/static-content
PORT=3001
REMOTE_URLS={ "mfe_meganav": "http://localhost:3002" }