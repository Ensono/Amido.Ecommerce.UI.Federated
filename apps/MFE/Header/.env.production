ASSETS_PATH=//localhost:3001
GENERATE_SOURCEMAP=false
PORT=3001
REMOTE_URLS={}
LOG_LEVEL=info
