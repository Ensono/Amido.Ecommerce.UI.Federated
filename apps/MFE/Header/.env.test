ASSETS_PATH=//localhost:3001
GENERATE_SOURCEMAP=true
PORT=3001
REMOTE_URLS={ "mfe_text_positioner": "http://localhost:3004" }
