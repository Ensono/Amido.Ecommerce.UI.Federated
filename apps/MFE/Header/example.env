PORT=3001
REMOTE_URLS={ "webpackRemote2": "http://localhost:3003" }