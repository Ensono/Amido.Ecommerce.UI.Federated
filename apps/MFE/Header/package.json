{
  "name": "@next/header",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.0.3",
    "@types/node": "^16.11.13",
    "@types/react": "^17.0.37",
    "@types/react-dom": "^17.0.11",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-scripts": "5.0.0",
    "typescript": "^4.5.4",
    "web-vitals": "^2.1.2"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --coverage --watchAll=false",
    "eject": "react-scripts eject",
    "lint": "eslint 'src/**/*.{jsx,js,tsx,ts}'",
    "docs": "typedoc --out ../../../docs/header src/*.*"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
