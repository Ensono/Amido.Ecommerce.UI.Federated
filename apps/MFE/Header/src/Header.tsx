import React from 'react'
import {useSelector} from 'react-redux'

import {federateComponent} from '@batman/federate-component'
import {getRemoteUrls} from '@batman/remote-urls'

const REMOTES = getRemoteUrls()

const TextPositioner = federateComponent('mfe_text_positioner', './text-positioner', REMOTES.mfe_text_positioner)

const Header: React.FC = ({children}) => {
  const count = useSelector(state => (state as any).counter.value)
  return (
    <div>
      <div>
        <p>Count: {count}</p>
      </div>
      <TextPositioner
        loadingFallback={<div>loading text positioner...</div>}
        errorFallback={<div>error loading text positioner!</div>}
        position="bottom"
      >
        {children}
      </TextPositioner>
    </div>
  )
}

export default Header
