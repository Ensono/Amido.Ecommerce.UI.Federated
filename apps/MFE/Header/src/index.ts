import dotenv from 'dotenv'

import server from './server'
import {Configuration} from './server/models'

const config = dotenv.config({path: `./.env.${process.env.NODE_ENV}`}) as Configuration
const PORT = config.PORT || '3001'
server.listen(PORT, () => console.log(`webpack host: started at http://localhost:${PORT}`))
