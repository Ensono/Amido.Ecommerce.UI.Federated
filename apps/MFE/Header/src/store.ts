import { configureStore } from "@reduxjs/toolkit";
import headerCounterReducer, { headerCounterSlice } from "./ducks/headerCounter";

export const headerStore = configureStore({
    reducer: {
        headerCounter: headerCounterReducer,
    },
})

export const headerCounterActions = headerCounterSlice.actions

export type RootState = ReturnType<typeof headerStore.getState>
export type AppDispatch = typeof headerStore.dispatch