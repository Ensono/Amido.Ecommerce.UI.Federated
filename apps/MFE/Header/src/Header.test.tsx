import {Provider} from 'react-redux'

import {render, waitFor} from '@testing-library/react'

import Header from './Header'
import {headerStore} from './store'

jest.mock('@batman/federate-component', () => {
  return {
    __esModule: true,
    federateComponent: ({remote}: any) =>
      function mockComponent() {
        return <section>{remote}</section>
      },
  }
})

describe('Header', () => {
  it('renders as expected', async () => {
    const {container} = render(
      <Provider store={headerStore}>
        <Header>foo bar</Header>
      </Provider>,
    )
    await waitFor(() => {
      expect(container).toMatchSnapshot()
    })
  })
})
