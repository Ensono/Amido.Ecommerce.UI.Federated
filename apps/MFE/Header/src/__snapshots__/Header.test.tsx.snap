// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expected 1`] = `
<div>
  <div>
    <div>
      <h2>
        Header Count: 
        0
      </h2>
      <button>
        Make number go up plz
      </button>
      <button>
        Make number go down pretty plz
      </button>
    </div>
    <div>
      loading text positioner...
    </div>
  </div>
</div>
`;
